packages/authentication/modules/user/useUser/index.ts

```diff
@@ -1,16 +1,28 @@
-import { COOKIE_NAME } from '@baseapp-frontend/utils'
+import { ACCESS_COOKIE_NAME, decodeJWT } from '@baseapp-frontend/utils'
 
 import { useQuery, useQueryClient } from '@tanstack/react-query'
 import Cookies from 'js-cookie'
 
 import UserApi, { USER_API_KEY } from '../../../services/user'
 import { IUser } from '../../../types/user'
-import { IUseUser } from './types'
+import { IUseSimpleTokenUser, IUseUser } from './types'
 
-const useUser = <TUser extends Partial<IUser>>({
+const useUser = ({ cookieName = ACCESS_COOKIE_NAME }: IUseUser = {}): IUser | null => {
```

Comment:
Could we still use the generics here? That gets pretty useful when projects change the User model and we can use the correct interface on it.
```suggestion
const useUser = <TUser extends Partial<IUser>, IJwtTokenSpecifics> ({ cookieName = ACCESS_COOKIE_NAME }: IUseUser = {}): IUser | null => {
```